crates/compiler/src/moo.pest

```diff
@@ -59,7 +59,7 @@ anycode = { ^"any" }
 return_statement = { ^"return" ~ (expr)? ~ ";" }
 expr_statement   = { (expr)? ~ ";" }
 
-expr = { prefix* ~ primary ~ postfix* ~ (infix ~ prefix* ~ primary ~ postfix*)* }
+expr = { (integer | (prefix* ~ primary)) ~ postfix* ~ (infix ~ (integer | (prefix* ~ primary)) ~ postfix*)* }
```

Comment:
Before this change the `-` would go into the `neg` rule, so we'd try to parse `-i64::MIN` into `i64` which doesn't fit into `i64`. So: need to parse negative literals as actual literals, instead of applying `neg` to the positive version.